src/Materials/PBR/babylon.pbrBaseMaterial.ts

```diff
@@ -686,15 +686,303 @@
                 subMesh._materialDefines = new PBRMaterialDefines();
             }
 
-            var scene = this.getScene();
-            var defines = <PBRMaterialDefines>subMesh._materialDefines;
+            const scene = this.getScene();
+            const engine = scene.getEngine();

```

Comment:
Could be move back after the check (agree it is preoptimization but happen per submesh per frame)

Reply:
Yup.